paz/processors/standard.py

```diff
@@ -485,6 +504,107 @@ def call(self, dictionary):
+        value: Float.
+
+    # Returns
+        Boolean.
+    """
+    def __init__(self, threshold=0.5):
+        super(FloatToBoolean, self).__init__()
+        self.threshold = threshold
+
+    def call(self, value):
+        return value > self.threshold
+
+
+class NoneConverter(Processor):
+    """Converts a None value to the last valid or a default value.
+    # Arguments
+        default_value: Any. Default value to convert to until a first valid value is stored.
+        value: Any Noneable value.
+
+    # Returns
+        Any.
+    """
+
+    def __init__(self, default_value=0.0):
+        super(NoneConverter, self).__init__()
+        self.default_value = default_value
+
+    def call(self, value):
+        if value is not None:
+            self.default_value = value
+        return self.default_value
+
+
+class AveragePredictions(Processor):
+    """Averages the last n predictions
+    # Arguments
+        window_size: Int. Number of predictions to average over.
+        value: Bool, Int or Float value. Value to average over.
+    # Returns
+        Bool, Int or Float value. Averaged value.
+    """
+
+    def __init__(self, window_size=1):
+        super(AveragePredictions, self).__init__()
+        if window_size <= 0:
+            raise ValueError('``window_size`` must be greater than 0')
+        self.window_size = window_size
+        self.predictions = []
+
+    def call(self, value):
+        if value is None:
+            mean = None
+        else:
+            self.predictions.append(value)
+            if len(self.predictions) > self.window_size:
+                self.predictions.pop(0)
+            mean = np.mean(self.predictions, axis=0)
+        return mean
+
+
+class WeightedAveragePredictions(Processor):
+    """Weighted averages the last n predictions
+    # Arguments
+        window_size: Int. Number of predictions to average over.
+        value: Int or Float value. Value to average over.
+    # Returns
+        Int or Float value. Averaged value.
+    """
+
+    def __init__(self, window_size=1):
+        super(WeightedAveragePredictions, self).__init__()
+        self.window_size = window_size
+        self.predictions = []
+
+    def call(self, value):
+        if value is None:
+            mean = None
+        else:
+            size = len(self.predictions)
+
+            self.predictions.append(value)
+            if size > self.window_size:
+                self.predictions.pop(0)
+
+            mean = 0
+            if len(self.predictions) <= 1:
+                mean = value
+            else:
+                total_weights = 0
+                for prediction_index in range(0, size):
+                    weight = (prediction_index + 1) / size
+                    mean = mean + self.predictions[prediction_index] * weight
+                    total_weights = total_weights + weight
+                mean = mean / total_weights
+        return mean
+      
+      
```

Comment:
I think this can be split into smaller functions to improve readability. The 3 conditions + for loop in a call seems to complex to read and understand easily.

Reply:
I have combined the AveragePredictions class with the WeightedAveragePredictions class. Additionally, I moved the computation of the weighted average (the for loop) to the backend.

For me it looks more readable now and I could save duplicated lines :)